src/components/projectSelect/index.tsx

```diff
@@ -83,3 +94,24 @@ export const ProjectSelect: React.FC<ProjectSelectProps> = ({
     />
   );
 };
+
+export const getProjects = (isProjectSettingsPage) => {
+  if (!isProjectSettingsPage) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { data, loading } = useQuery<
+      GetProjectsQuery,
+      GetProjectsQueryVariables
+    >(GET_PROJECTS);
+    const { projects } = data || { projects: [] };
+
+    return { projects, loading };
+  }
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { data, loading } = useQuery<
+    GetViewableProjectRefsQuery,
+    GetViewableProjectRefsQueryVariables
+  >(GET_VIEWABLE_PROJECTS);
+  const { viewableProjectRefs } = data || { viewableProjectRefs: [] };
+  // const projects = ;
+  return { projects: viewableProjectRefs.map((p) => ({ ...p })), loading };
+};
```

Comment:
You can't actually do this. Conditional hooks are not allowed because they break reacts ability to track the hooks. 
https://reactjs.org/docs/hooks-rules.html
The rules-of-hooks rule is there to prevent this. `useQuery` has a [param called skip](https://www.apollographql.com/docs/react/data/queries/#skip) you can use instead

Reply:
fixed, thanks!